libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1313,45 +1370,14 @@ void IotMqtt_Disconnect( IotMqttConnection_t mqttConnection,
 
             if( status == IOT_MQTT_SUCCESS )
             {
-                /* Check the serialized MQTT packet. */
-                IotMqtt_Assert( pOperation->u.operation.pMqttPacket != NULL );
-                IotMqtt_Assert( pOperation->u.operation.packetSize > 0 );
-
-                /* Schedule the DISCONNECT operation for network transmission. */
-                if( _IotMqtt_ScheduleOperation( pOperation,
-                                                _IotMqtt_ProcessSend,
-                                                0 ) != IOT_MQTT_SUCCESS )
-                {
-                    IotLogWarn( "(MQTT connection %p) Failed to schedule DISCONNECT for sending.",
-                                mqttConnection );
-                    _IotMqtt_DestroyOperation( pOperation );
-                }
-                else
-                {
-                    /* Wait a short time for the DISCONNECT packet to be transmitted. */
-                    status = IotMqtt_Wait( pOperation,
-                                           IOT_MQTT_RESPONSE_WAIT_MS );
-
-                    /* A wait on DISCONNECT should only ever return SUCCESS, TIMEOUT,
-                     * or NETWORK ERROR. */
-                    if( status == IOT_MQTT_SUCCESS )
-                    {
-                        IotLogInfo( "(MQTT connection %p) Connection disconnected.", mqttConnection );
-                    }
-                    else
-                    {
-                        IotMqtt_Assert( ( status == IOT_MQTT_TIMEOUT ) ||
-                                        ( status == IOT_MQTT_NETWORK_ERROR ) );
-
-                        IotLogWarn( "(MQTT connection %p) DISCONNECT not sent, error %s.",
-                                    mqttConnection,
-                                    IotMqtt_strerror( status ) );
-                    }
-                }
+                /* Processing the operation after sending it on the network. */
+                _IotMqtt_ManagedMqttSend( pOperation );
             }
             else
             {
-                EMPTY_ELSE_MARKER;
+                IotLogWarn( "(MQTT connection %p) Failed to schedule DISCONNECT for sending.",
```

Comment:
Are we using taskpool schedule still with this change? I think we don't. If that is the case, please update the comment as well to indicate that